src/server/services/wfs/qgswfsgetfeature.cpp

```diff
@@ -1313,10 +1319,19 @@ namespace QgsWfs
         }
         else
         {
-          const QgsAbstractGeometry *abstractGeom = geom.constGet();
-          if ( abstractGeom )
+          if ( params.forceGeomToMulti && ! QgsWkbTypes::isMultiType( geom.wkbType() ) )
+          {
+            QgsGeometry multiGeom = geom;
+            multiGeom.convertToMultiType();
+            gmlElem = QgsOgcUtils::geometryToGML( multiGeom, doc, prec );
```

Comment:
Why can't asGml2 be used in this case?

Reply:
because I have cloned the upper line like centroid.